backend/perf.js

```diff
@@ -80,275 +79,285 @@ function init(repo_loc, callback) {
         });
 }
 
-function read_files(repo_loc) {
+function read_files(repo_loc, callback) {
     // Read all files from repo_loc/processed
-    fs.readdir(path.resolve(__dirname, repo_loc, 'processed'), function (err, files) {
-        if (err) {
-           throw err;
+    let files = fs.readdirSync(path.resolve(__dirname, repo_loc, "processed"));
+
+    let no_times = 0;
+    let c_bootstraps = 0;
+    let c_benchmarks = 0;
+    let c_benchmarks_add = 0;
+    for (let filename of files) {
+        let full_name = path.resolve(__dirname, repo_loc, "processed", filename);
+        let file_contents = fs.readFileSync(full_name, "utf8");
+        if (file_contents.length === 0) {
+            no_times += 1;
+            continue;
         }
-        var no_times = 0;
-        var c_bootstraps = 0;
-        var c_benchmarks = 0;
-        var c_benchmarks_add = 0;
-        files.map(function (filename) {
-            var full_name = path.resolve(__dirname, repo_loc, 'processed', filename);
-            var file_contents = fs.readFileSync(full_name, 'utf8');
-            if (!file_contents) {
-                no_times += 1;
-                return;
-            }
-            var contents = JSON.parse(file_contents);
-            var header = contents.header;
-            var times = contents.times;
-            if (times.length == 0) {
-                no_times += 1;
-                return;
+        let contents = JSON.parse(file_contents);
+        let header = contents.header;
+        let times = contents.times;
+        if (times.length == 0) {
+            no_times += 1;
+            continue;
+        }
+        let test_name = filename.substring(0, filename.indexOf("--"));
+
+        let date = new Date(header.date);
+        if (isNaN(date)) {
+            let date_components = [0, 0, 0, 0, 0, 0];
+            let actual = header.date.split("-").map(comp => parseInt(comp, 10));
+
+            for (let i = 0; i < actual.length; i++) {
```

Comment:
Either works, but the above makes it clear that it is an array. I would've used for-of syntax, but we need the index to update `date_components`. Let me know if you prefer for-in. 

I also think that at least to some extent, the JS community is moving away from for-in syntax in preference to for-of, c-style for, and for-of with Object.keys. There is also [this](http://stackoverflow.com/questions/500504/why-is-using-for-in-with-array-iteration-a-bad-idea) to consider, although most of those concerns are probably not applicable in this specific case.